src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
 		}
 		configureDriver();
 		return driver;
-    }
+	}
+
+	public void initWithReporter(TestRunReporter reporter) {
+		reporter.addListener(this);
+	}
+
+	@Override
+	public void reported(SemanticUnit unit, Action action, String msg) {
+		if (unit == SemanticUnit.TEST && action == Action.ENTER) {
+			runningTest = msg;
+		}
+		if (screenshotShouldBeMade(unit, action, msg)) {
+			screenshot(msg + '.' + action.name());
+		}
+	}
+
+	private String runningTest = null;
+	private static final int SCREENSHOT_FILENAME_MAXLEN = 128;
+
+	private String getCurrentTestCase() {
+		return runningTest != null ? runningTest : "UNKNOWN_TEST";
+	}
+
+	private String getScreenshotPath() {
+		// configurable through maven build?
```

Comment:
will hopefully be implemented with https://testeditor.atlassian.net/browse/TE-472